vendor/k8s.io/kubernetes/pkg/kubectl/cmd/patch.go

```diff
@@ -185,8 +188,20 @@ func RunPatch(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 			}
 			count++
 
+			oldData, err := json.Marshal(info.Object)
+			if err != nil {
+				return err
+			}
+			newData, err := json.Marshal(patchedObj)
+			if err != nil {
+				return err
+			}
+			if !reflect.DeepEqual(oldData, newData) {
```

Comment:
@fabianofranz Thanks, went ahead and opened an upstream PR for this here:  https://github.com/kubernetes/kubernetes/pull/38538